Source/WebCore/page/Page.cpp

```diff
@@ -3174,6 +3175,13 @@ void Page::setActivityState(OptionSet<ActivityState> activityState)
         performanceMonitor->activityStateChanged(oldActivityState, activityState);
 }
 
+void Page::allowHDR(bool allow)
+{
+    RefPtr localMainFrame = this->localMainFrame();
+    if (RefPtr view = localMainFrame ? localMainFrame->view() : nullptr)
```

Comment:
This only does the main frame. We'll need to traverse into subframes.  This will also need extra work for site isolation.